### PR TITLE
SVA-185 - iOS Release Pipeline

### PR DESCRIPTION
**Pull request by jppferguson**
- [Description](#description)
  - [Type of change](#type-of-change)
- [Checklist](#checklist)

# Description

Setup fastlane to make deployments of the iOS app from a single command - `fastlane beta` (from the ios directory).

As part of this I had to allow connecting to the API over HTTP as we don't have a certificate for the API yet as iOS requires you to connect to all URLs over HTTPS unless explicitly excluded (in the following file: `app/ios/STA/Info.plist`). Once we are able to connect over HTTPS we should remove this.

Fixes # SVA-185

## Type of change

Please delete options that are not relevant
- [x] New Feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (if required)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/ios/STA/Info.plist [hunk 1]
```diff
@@ -21,7 +21,7 @@
 	<key>CFBundleSignature</key>
 	<string>????</string>
 	<key>CFBundleVersion</key>
-	<string>1</string>
+	<string>3</string>
```
Comment (joefhall) [inline app/ios/STA/Info.plist:24]: Should this be `4` now if it should match the `versionCode` [here](https://github.com/Scottish-Tech-Army/Volunteer-app/blob/main/app/android/app/build.gradle#L132)?

File: app/ios/STATests/Info.plist [hunk 2]
```diff
@@ -19,6 +19,6 @@
 	<key>CFBundleSignature</key>
 	<string>????</string>
 	<key>CFBundleVersion</key>
-	<string>1</string>
+	<string>3</string>
```
Comment (joefhall) [inline app/ios/STATests/Info.plist:22]: Should this be `4` now if it should match the `versionCode` [here](https://github.com/Scottish-Tech-Army/Volunteer-app/blob/main/app/android/app/build.gradle#L132)?

**Review by joefhall — CHANGES_REQUESTED** [review]
Nice one, thanks @jppferguson!  Just a few small suggestions/questions.

I don't have a Mac but hopefully someone who does can also review to see if it looks like this is working for them locally.

File: README.md [hunk 3]
```diff
@@ -210,4 +210,12 @@ For support, please @ David Calder in the [volunteer-app](https://scottishtechar
 
 9. In `/app/src/Config/index.ts` set `STA_BASE_URL` to point to **the IP address** for the external URL for the API endpoint on AWS, see [known issues](#known-issues) above -- not to your localhost or its IP address.
```
[the remark this answers is not in the document]
Comment (jppferguson) [reply README.md:211]: Nice one, yeah I think this was sorted out in Rosie's Readme updates 👍🏻

File: app/ios/STA/Info.plist (hunk 1 above)
Comment (jppferguson) [reply app/ios/STA/Info.plist:24]: We can try to keep the Android and iOS build numbers in sync but in my experience, they usually tend to diverge a bit as we'll likely need to make separate builds for each platform. The semantic version we should definitely keep in sync though.

File: app/.env.example [hunk 4]
```diff
@@ -0,0 +1 @@
+# APPLE_ID=your.apple.id@email.com
```
[the remark this answers is not in the document]
Comment (jppferguson) [reply app/.env.example:1]: Yeah nice – good point!

File: README.md [hunk 5]
```diff
@@ -210,4 +210,12 @@ For support, please @ David Calder in the [volunteer-app](https://scottishtechar
...
 
+10. Go to the `/app/ios` directory in a terminal window and run the command `fastlane beta`.  You'll be prompted to login (maybe several times) with your App Store Connect login.
```
[the remark this answers is not in the document]
Comment (jppferguson) [reply README.md:213]: Cool, yeah I'll add to these too, completely missed them!

File: app/ios/STA/Info.plist (hunk 1 above)
Comment (joefhall) [reply app/ios/STA/Info.plist]: Ah right, cool -- that's handy to know! 👍

**Review by joefhall — APPROVED** [review]
Great stuff, thanks @jppferguson!

File: app/ios/STA/Info.plist (hunk 1 above)
Comment (rc-1988) [reply app/ios/STA/Info.plist]: In the draft [Android deployment PR](https://github.com/Scottish-Tech-Army/Volunteer-app/pull/84) that I'm working on, I'm currently taking the app/package.json to be the single source of truth as to which semantic version we're up to.

Is the semantic version also needed in the iOS deployment process, and if so, where does it get it from? Does pulling it from package.json work for keeping it in sync across the two platforms?

**Review by PammyLo — APPROVED** [review]
Looks great, thanks @jppferguson!

**Pull request merged**